patchlab/gitlab2email.py

```diff
@@ -85,6 +85,18 @@ def email_merge_request(
             )
             return
 
+    # If the merge request head SHA matches a prior bridged submission, skip it.
+    prior_sub = BridgedSubmission.objects.filter(
```

Comment:
There's a race condition here where two events on a merge request can run concurrently, both pass this prior submission check, and then send out either duplicate emails or a vN and vN+1 in close succession.

One fix might be to be choosier about the merge events we accept as triggers (e.g. don't run a task if a tag got added, only look at push related events), but it would also probably be better to check just before sending the emails to shorten the race window. It's probably not worth introducing locking since the failure case isn't catastrophic.